lib/repl.js

```diff
@@ -1848,6 +1844,8 @@ function defineDefaultCommands(repl) {
 }
 
 function Recoverable(err) {
+  if (!(this instanceof Recoverable))
+    return deprecateInstantiation(Recoverable, 'DEP0185', err);
```

Comment:
Before this PR, it was not possible to construct `Recoverable` without `new`, it seems weird to add it while deprecating it
```suggestion
```